test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -2075,27 +2072,27 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
             var largeString = new string('a', maxBytesPreCompleted + 1);
 
             var writeTcs = new TaskCompletionSource<object>();
-            var registrationWh = new ManualResetEventSlim();
-            var connectionCloseWh = new ManualResetEventSlim();
+            var registrationWh = new TaskCompletionSource<object>();
+            var connectionCloseWh = new TaskCompletionSource<object>();
```

Comment:
The test was hanging indefinitely on `.Wait()`. I switched to tcs so I could add .TimeoutAfter assertions.